### ISSUE TITLE
[AIR] Support postprocessing in Predictors 

### ISSUE THREAD

**Issue by amogkam** [ray-project/ray#24979]
### Description

Often times, there is additional postprocessing that needs to be done after the model outputs. For example, this could include converting logits outputted by the model to a predicted class or converting a sequence of numbers to a sentence.

Currently, the only way to support this postprocessing **uniformly for both offline and online inference** is to subclass and override predictors.

Instead, we should allow postprocessing functionality to be pluggable into predictors and also provide some useful off the shelf postprocessors.

This has been brought up by multiple users: https://docs.google.com/document/d/1yub1d3ck0H7Zpb7vgzwgFo3t_E4oYWvtugG4qYoOgBU/edit#

**Comment by cszhu** [comment]
Hello! This P2 issue has seen no activity in at least 2 years. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
